src/sentry/data_export/utils.py

```diff
@@ -50,16 +50,20 @@ def wrapped(*args, **kwargs):
                     (
                         snuba.RateLimitExceeded,
                         snuba.QueryMemoryLimitExceeded,
+                        snuba.QueryExecutionTimeMaximum,
```

Comment:
For QueryMemoryLimitExceeded, QueryExecutionTimeMaximum and QuerySizeExceeded, it might be a better experience to suggest that the user try exporting less data (fewer days, fewer rows).

Reply:
The number of rows per query is out of the control of the users here. We can try automatically decreasing the limit parameter here upon retries to accommodate that, but that would be out of scope of this PR.